### PR TITLE
Reduced allocations in  ShiftedJetProducerT

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Reduced allocations within jet correction uncertainties conditions
- Improved ShiftedJetProducerT
   - reduced allocations
   - added fillDescriptions
   - modernized C++ syntax
   - updated to newer framework syntax
   - only update data derived from the EventSetup if the EventSetup data changes

#### PR validation:

Code compiles. A simple job using a prototype alloc monitor showed reduced amount of memory held from event to event.

NOTE: this should just be a refactoring so the PR tests should show no differences. If there are differences than debugging will be necessary.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
The differences in the PR test appear to be unrelated to this PR (they are from HGCal which is known to produce inconsistent results).

**Comment by hqucms** [comment]
+1

File: CondFormats/JetMETObjects/src/JetCorrectionUncertainty.cc [hunk 1]
```diff
@@ -69,23 +68,19 @@ JetCorrectionUncertainty::JetCorrectionUncertainty(const JetCorrectorParameters&
   mIsLepPyset = false;
   mIsLepPzset = false;
   mAddLepToJet = false;
-  mUncertainty = new SimpleJetCorrectionUncertainty(fParameters);
 }
 /////////////////////////////////////////////////////////////////////////
-JetCorrectionUncertainty::~JetCorrectionUncertainty() { delete mUncertainty; }
-/////////////////////////////////////////////////////////////////////////
 void JetCorrectionUncertainty::setParameters(const std::string& fDataFile) {
   //---- delete the mParameters pointer before setting the new address ---
```
Comment (perrotta) [inline CondFormats/JetMETObjects/src/JetCorrectionUncertainty.cc:74]: This comment is not needed any more, and it can be misleading
```suggestion
```

Comment (Dr15Jones) [reply CondFormats/JetMETObjects/src/JetCorrectionUncertainty.cc]: removed

**Comment by Dr15Jones** [comment]
please test 

@hqucms @jfernan2 please resign as the only change was the removal of a comment.

**Comment by perrotta** [comment]
+db

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**